src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs

```diff
@@ -76,6 +77,15 @@ public class DownloadPipelineArtifactTaskV2_0_0 : PipelineArtifactTaskPluginBase
             string projectName = context.GetInput(ArtifactEventProperties.Project, required: false);
             string tags = context.GetInput(ArtifactEventProperties.Tags, required: false);
             string userSpecifiedpipelineId = context.GetInput(BuildId, required: false);
+            string defaultWorkingDirectory = context.Variables.GetValueOrDefault("system.defaultworkingdirectory").Value;
+
+            targetPath = Path.IsPathFullyQualified(targetPath) ? targetPath : Path.GetFullPath(Path.Combine(defaultWorkingDirectory, targetPath));
```

Comment:
I think this should be, if it's a file path, throw, if it's a directory, we should create it if it doesn't exist, similar to V1.

Reply:
@willsmythe  As Tanmay mentioned, for the current relative path support, if the directory doesn't exist then we'll throw file not found exception. If we remove the check 
!Directory.Exists(targetPath)  - it should create missing directories automatically - 
string fullPath = this.CreateDirectoryIfDoesntExist(targetPath);

Reply:
Question is should we create directories if the specified directories don't exist.

Reply:
Yah, we should always try to create the target path, regardless if the user specified an absolute or relative path for the "path" input.